model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1087,6 +1087,44 @@ message SideInput {
   FunctionSpec window_mapping_fn = 3;
 }
 
+message StandardArtifacts {
+  enum Types {
+    // A URN for artifacts stored in a local directory.
+    // payload: ArtifactFilePayload.
+    FILE     = 0 [(beam_urn) = "beam:artifact:file:v1"];
+    // A URN for artifacts embedded in ArtifactInformation proto.
+    // payload: raw data bytes.
+    EMBEDDED = 1 [(beam_urn) = "beam:artifact:embedded:v1"];
+    // A URN for artifacts described by HTTP links.
+    // payload: a string for an artifact HTTP URL
+    HTTP     = 2 [(beam_urn) = "beam:artifact:http:v1"];
+    // A URN for artifacts hosted on PYPI.
+    // artifact_id: a PYPI project name
+    // version_range: a PYPI compatible version string
+    // payload: None
+    PYPI     = 3 [(beam_urn) = "beam:artifact:pypi:v1"];
+    // A URN for artifacts hosted on Maven central.
+    // artifact_id: [maven group id]:[maven artifact id]
+    // version_range: a Maven compatible version string
+    // payload: None
+    MAVEN    = 4 [(beam_urn) = "beam:artifact:maven:v1"];
+  }
+}
+
+message ArtifactFilePayload {
+  // A path to an artifact file on a local system.
+  string local_path = 1;
```

Comment:
It's assumed to be used when SDK is saving remote artifacts and preparing to submit them to the staging service from locally downloaded files. `local_path` here is not ubiquitous information. After submitting the artifacts, retrieval_token (and staged_name if necessary) should be used instead of `local_path`. Is there any specific use-cases you're considering?

Reply:
One issue here is that the callee may not know if it is on a shared filesystem with the caller (e.g. when calling the expansion service). And when calling two distinct expansion services, one would like to be able to compare between them. 

Also, perhaps we should not limit ourselves to local paths here, but any path that can be opened with beam filesystems.

Reply:
How about just `path` and we drop `staged_name`?

OR rename ArtifactFilePayload to ArtifactUrlPayload and then users can reference local and remote artifacts?
Using a `url` field which would always have the scheme specified (e.g. local file paths would always have `file://` prefix instead of the assumed conversion from `/foo/bar` to `file:///foo/bar`